sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -154,9 +154,18 @@ public void writeThenReadAll() {
 
     PipelineResult result = pipeline.run();
     PipelineResult.State pipelineState = result.waitUntilFinish();
-    assertEquals(
-        Lineage.query(result.metrics(), Lineage.Type.SOURCE),
-        Lineage.query(result.metrics(), Lineage.Type.SINK));
+
+    // FileIO Lineage has multi-level cap to avoid huge set. Only check both non-empty
+    // (if supported by runner) for large shards
+    if (numShards <= 100) {
+      assertEquals(
+          Lineage.query(result.metrics(), Lineage.Type.SOURCE),
+          Lineage.query(result.metrics(), Lineage.Type.SINK));
+    } else {
+      assertEquals(
+          Lineage.query(result.metrics(), Lineage.Type.SOURCE).isEmpty(),
```

Comment:
In case of V1 wouldn't this just test equality of F == F and later after service side change for v2 same?

Reply:
Thanks for pointing this out. After review I think we can keep TextIOIT untouched, let me testing with TextIOIT-manyfiles test

Reply:
confirm that this does not need to change. Ran TextIOIT-manyfiles and test passed the original assert